### DISCUSSION TITLE
p.262 apply_adminの言葉の意図が不明

### DISCUSSION THREAD

**Discussion by ac0mz** [budougumi0617/go_todo_app#45] [Q&A] [answered]
「ミドルウェアの適用順序に気をつけながら実装したのが`apply_admin`です。」とありますが、`apply_admin`がどういったものかは書籍上で読み取れませんでした。
またサンプルリポジトリを検索しても上記単語は出てきません。
本来`apply_admin`として表現したかった内容が何か、教えて頂ければ幸いです。

**Comment by budougumi0617** [comment, marked as answer]
ご指摘ありがとうございます。書籍内の誤りについて複数ご報告いただき大変助かりました。ありがとうございました。

本件については、　`/admin` エンドポイント（リスト20.47）のミドルウェアの適応順序について言及したかった文でした。
不足分の説明も含めると次のような文に修正・追記するのがよいかなと思いました。

> ミドルウェアの適用順序に気をつけながら実装したのがリスト20.47の実装です。リスト20.47では、アクセストークンからhttp.Request型の値に含まれるcontext.Context型の値にユーザーIDとロールを埋め込むhandler.AuthMiddlewareが実行されたあとにhandler.AdminMiddlewareが実行され順序でミドルウェアを適用しています。

正誤表に修正・追記を記載いたしました。
https://github.com/budougumi0617/go_todo_app/releases/tag/v1.0.5
https://github.com/budougumi0617/go_todo_app/blob/v1.0.5/errata.md

**Reply by ac0mz** [reply]
ご回答ありがとうございます。ご説明の内容で理解いたしました。

> handler.AdminMiddlewareが実行され順序でミドルウェアを適用しています。

上記の「〜が実行され順序で」は誤字のように見受けられます。
代替案としては「handler.AdminMiddlewareが実行される順序でミドルウェアを適用しています。」でいかがでしょうか。
もしくは「handler.AdminMiddlewareを実行する順序でミドルウェアを適用しています。」も自然かと思います。

**Reply by budougumi0617** [reply]
修正後の文書までレビューしていただきありがとうございます。
長めの一文で全体的に読みにくいなと感じたので指摘部分以外も修正してみました。

https://github.com/budougumi0617/go_todo_app/compare/v1.0.5...v1.0.6

**Reply by ac0mz** [reply]
自然な文章でとても分かりやすいです！

この本は補足も含めて有益な情報の量が多いので、何周もして腹落ちさせる価値のある内容だと思っています。
素敵な本を出版していただき感謝です！

**Reply by budougumi0617** [reply]
こんなに間違いを見つけていただいたあとにそう言っていただき大変恐縮です。
今月は頂いたありがたいお言葉を胸に生きていきます 😭 !!
